pkg/apply/taskrunner/task.go

```diff
@@ -218,10 +239,15 @@ func (w *WaitTask) StatusUpdate(taskContext *TaskContext, id object.ObjMetadata)
 	switch {
 	case w.pending.Contains(id):
 		// pending - check if reconciled
-		if w.reconciledByID(taskContext, id) {
+		if w.reconciledByID(taskContext, id) { //nolint:gocritic
```

Comment:
just use a switch statement

Reply:
Done. Open to ways we can simplify this to avoid nested switch statements, but it isn't too bad.

Reply:
you could extract the inner switch to a function, but only if it makes it easier to read with a good name.

Reply:
Yeah, I think leaving it the way it is might be better for now. Having the logic in one place makes it easier to see what is going on.